src/components/welcome/WelcomeLogin/index.tsx

```diff
@@ -17,22 +17,29 @@ const WelcomeLogin = () => {
   const { isLoaded, hasSafes } = useHasSafes()
   const [shouldRedirect, setShouldRedirect] = useState(false)
 
+  const redirect = useCallback(() => {
+    if (wallet) {
+      if (isLoaded && !hasSafes) {
+        trackEvent(CREATE_SAFE_EVENTS.OPEN_SAFE_CREATION)
+        router.push({ pathname: AppRoutes.newSafe.create, query: router.query })
+      } else {
+        router.push({ pathname: AppRoutes.welcome.accounts, query: router.query })
+      }
+    }
+  }, [hasSafes, isLoaded, router, wallet])
+
   const onLogin = useCallback(() => {
     setShouldRedirect(true)
   }, [])
 
+  const onContinue = useCallback(() => {
+    redirect()
+  }, [redirect])
```

Comment:
```suggestion
  const onContinue = useCallback(() => {
    redirect()
  }, [redirect])
```

Essentially it's the same as `const onContinue = redeirect`. So you could just do `onContinue={redirect}` in the JSX below.